app/models/user.rb

```diff
@@ -14,4 +17,12 @@ def normalize_queue_item_positions
   def queued_item?(item)
     queue_items.map(&:video).include?(item)
   end
+
+  def follows?(another_user)
+    following_relationships.map(&:leader).include?(another_user)
+  end
+
+  def can_follow?(another_user)
+    (self == another_user || !self.follows?(another_user))
```

Comment:
This is actually wrong hm, should be `self != another_user` because he can follow someone else not himself. or the bang should be before the braces like this `!(self == another_user || self.follows?(another_user))`